explainer.md

```diff
@@ -1,624 +1,397 @@
-(MPA). The SPA equivalent of this case is one where the old document is mutated
-into the new document via DOM APIs. See [API Extensions](#api-extensions) for
-more code examples.
+The goal is to provide a mechanism and API which will allow simple transitions like above to be specified in CSS, building on CSS animations, but also allow for more complex transition to be performed via JavaScript, building on the Web Animations API.
 
-Note that the API shape below is tentative and used to explain the core feature
-design.
+This section covers the concepts and mechanisms, while a later section looks at possible API shapes.
 
-### Old Document
+Performing a transition from Page-A to Page-B requires parts of both to be on screen at the same time, potentially moving independently. This is currently impossible in a cross-document navigation, but it's still hard in an SPA (single page app) navigation. You need to make sure that the outgoing state can't receive additional interactions, and ensure the presence of both states doesn't create a confusing experience for those using accessibility technology.
 
-```html
-<html>
-<head>
-  <style>
-    body {
-      background-color: blue;
-    }
-    .animated {
-      position: relative;
-      top: 300px;
-      width: 100px;
-      height: 100px;
-      background-color: red;
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-  <a href="new-document.html">Click Me</a>
-</body>
-</html>
-```
+The aim of this design is to allow for representations of both Page-A and Page-B to exist at the same time, without the usability, accessibility, or memory concerns of having both complete DOM trees alive.
 
-### New Document
+Here's the example that will be used to explain the design:
 
-```html
-<html>
-<head>
-  <style>
-    body {
-      background-color: lightblue;
-    }
-    .animated {
-      position: relative;
-      top: 100px;
-      width: 200px;
-      height: 200px;
-      background-color: green;
-    }
-
-    ::shared-container(header-id) {
-      animation: ::shared-container-header-id 1s ease-in;
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-</body>
-</html>
+<img alt="Page-A and Page-B" src="media/pages.png?raw=true">
+
+The transition between Page-A and Page-B can be a full cross-document navigation between two same-origin pages, or it can be an SPA navigation. The main mechanism and concepts are the same between the two.
+
+Cross-origin transitions are something we want to tackle, but may have significant differences and restrictions for security reasons. Cross-origin transitions are not covered in this document.
+
+## Part 1: The offering
+
+Before Page-A goes away, it offers up parts of itself to be used in the transition. Generally, this will mean one part per 'thing' that acts independently during the transition. For the example transition, the parts are:
```

Comment:
Re-iterating from the internal discussion, the way we identify "parts" of a page that are cached and retained is through elements. So I find the "part" terminology unnecessarily abstract. We should just call them elements and it would make sense to anyone familiar with web elements.

We started with the word shared because I guess this was originally being used for transitions where we have matched elements from both ages. So they were sort of "shared" between the 2 pages. Also because we borrowed from the Android terminology of the same feature. I'd prefer to stick with shared for now since it's used consistently in all other documentation/code and we don't have any better ideas. If this comes up during external discussion, we can field suggestions and change it for once.